src/_bentoml_sdk/service/factory.py

```diff
@@ -207,12 +211,63 @@ def doc(self) -> str:
         return get_default_svc_readme(self)
 
     def schema(self) -> dict[str, t.Any]:
+        def _build_full_path(*path_segments: str) -> str:
+            # Combines URL path segments into a normalized path.
+            # Ensures it starts with '/' and has no trailing '/' (unless it's just '/').
+            # Collapses multiple slashes.
+            parts: list[str] = []
+            for segment in path_segments:
+                if not segment:
+                    continue
+                # Split by '/' and filter out empty strings that result from '//' or trailing/leading '/'
+                parts.extend(p for p in segment.split("/") if p)
+
+            if not parts:
+                return "/"
+
+            normalized_path = "/" + "/".join(parts)
+            return normalized_path
+
+        # Add API method routes (these are already full paths from method.route)
+        all_paths: dict[str, PathMetadata] = dict(
+            (_build_full_path(method.route), {"mounted": False})
+            for method in self.apis.values()
+        )
+
+        # Store id(app) to avoid reprocessing
+        processed_mounted_apps: set[int] = set()
+
+        def extract_routes_from_asgi_app(app: t.Any, current_prefix: str) -> None:
+            if app is None or id(app) in processed_mounted_apps:
+                return
+            processed_mounted_apps.add(id(app))
+
+            # Introspect ASGI app (FastAPI/Starlette)
+            if hasattr(app, "routes") and isinstance(app.routes, list):
```

Comment:
Maybe `isinstance(app, Starlette)` for better typing?

Reply:
updated